src/mono/mono/metadata/loader.c

```diff
@@ -787,14 +780,6 @@ mono_method_get_signature_checked (MonoMethod *method, MonoImage *image, guint32
 
 			sig = (MonoMethodSignature *)cache_memberref_sig (image, sig_idx, sig);
 		}
-		/* FIXME: we probably should verify signature compat in the dynamic case too*/
-		if (!mono_verifier_is_sig_compatible (image, method, sig)) {
-			guint32 klass = cols [MONO_MEMBERREF_CLASS] & MONO_MEMBERREF_PARENT_MASK;
```

Comment:
This one needs to stay too, it checks that the caller and callee assembly agree on the signature.